scala/src/com/google/idea/blaze/scala/run/producers/GenerateExecutableDeployableJarProviderTaskProvider.java

```diff
@@ -0,0 +1,181 @@
+import com.google.idea.blaze.base.command.BlazeCommandName;
+import com.google.idea.blaze.base.command.BlazeFlags;
+import com.google.idea.blaze.base.command.BlazeInvocationContext;
+import com.google.idea.blaze.base.command.buildresult.BuildResultHelper;
+import com.google.idea.blaze.base.console.BlazeConsoleLineProcessorProvider;
+import com.google.idea.blaze.base.filecache.FileCaches;
+import com.google.idea.blaze.base.model.primitives.Label;
+import com.google.idea.blaze.base.model.primitives.WorkspaceRoot;
+import com.google.idea.blaze.base.projectview.ProjectViewManager;
+import com.google.idea.blaze.base.projectview.ProjectViewSet;
+import com.google.idea.blaze.base.scope.BlazeContext;
+import com.google.idea.blaze.base.scope.Scope;
+import com.google.idea.blaze.base.scope.ScopedTask;
+import com.google.idea.blaze.base.settings.Blaze;
+import com.google.idea.blaze.base.util.SaveUtil;
+import com.intellij.execution.BeforeRunTask;
+import com.intellij.execution.BeforeRunTaskProvider;
+import com.intellij.execution.application.ApplicationConfiguration;
+import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.jar.JarApplicationConfiguration;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import icons.BlazeIcons;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+
+public class GenerateExecutableDeployableJarProviderTaskProvider
+  extends BeforeRunTaskProvider<GenerateExecutableDeployableJarProviderTaskProvider.Task> {
+  public static final Key<GenerateExecutableDeployableJarProviderTaskProvider.Task> ID = Key.create("CreateTempScalaBinaryTarget");
+
+  static class Task extends BeforeRunTask<Task> {
+    Task() {
+      super(ID);
+      setEnabled(true);
+    }
+  }
+
+  private final Project project;
+
+  public GenerateExecutableDeployableJarProviderTaskProvider(Project project) {
+    this.project = project;
+  }
+
+  @Override
+  public Key<Task> getId() {
+    return ID;
+  }
+
+  @Override
+  public Icon getIcon() {
+    return BlazeIcons.Blaze;
+  }
+
+  @Override
+  public Icon getTaskIcon(GenerateExecutableDeployableJarProviderTaskProvider.Task task) {
+    return BlazeIcons.Blaze;
+  }
+
+  @Override
+  public String getName() {
+    return "Generate executable deployable JAR for custom target";
+  }
+
+  @Override
+  public final boolean canExecuteTask(RunConfiguration configuration, GenerateExecutableDeployableJarProviderTaskProvider.Task task) {
+    return isValidConfiguration(configuration);
+  }
+
+  private boolean isValidConfiguration(RunConfiguration runConfiguration) {
+    return runConfiguration instanceof JarApplicationConfiguration;
```

Comment:
What determines whether this is a JarApplicationConfiguration? Should DeployableJarRunConfigurationProducer specify this explicitly, rather than supporting any ApplicationConfiguration type?

Reply:
@brendandouglas oops, this was a leftover from a previous attempt, where I tried using the JarApplication dialog. I'll update it, it should be `ApplicationConfiguration` instead! Thanks!